### PR TITLE
Trigger unfold on double click

### PR DESCRIPTION
**Pull request by fcollonval**

### CODE REVIEW

**Comment by martinRenou** [comment]
Can you provide more context? The single-click behavior was intentionally implemented in https://github.com/martinRenou/jupyterlab-unfold/pull/7

**Comment by martinRenou** [comment]
Making it a configuration setting, and rebased

File: src/unfold.ts [hunk 1]
```diff
@@ -220,21 +225,29 @@ export class DirTreeListing extends DirListing {
...
+        this.model.path = '/' + PathExt.dirname(entry.path);
       }
     }
   }
 
+  private async _eventDblClick(event: MouseEvent): Promise<void> {
+    const entry = this.modelForClick(event);
+
+    if (entry?.type === 'directory') {
+      this.model.toggle(entry.path);
```
Comment (fcollonval) [inline src/unfold.ts:247]: Do we want to keep that action if `singleClickToUnfold` is `true`?

**Review by fcollonval — COMMENTED** [review]
Hey @martinRenou, thanks for taking over. I added some comments.

File: src/index.ts [hunk 2]
```diff
@@ -30,14 +34,20 @@ const namespace = 'filebrowser';
...
   activate: async (
     app: JupyterFrontEnd,
     docManager: IDocumentManager,
     translator: ITranslator,
+    settings: ISettingRegistry,
     state: IStateDB | null
   ): Promise<IFileBrowserFactory> => {
+    const setting = await settings.load(SETTINGS_ID);
+
+    const singleClickToUnfold = setting.get('singleClickToUnfold')
```
[the remark this answers is not in the document]
Comment (martinRenou) [reply src/index.ts:48]: Thanks for your comment 😊 I'll do that

File: src/unfold.ts (hunk 1 above)
Comment (martinRenou) [reply src/unfold.ts:247]: I guess we should remove it

**Pull request merged**